src/os/bluestore/BtreeAllocator.cc

```diff
@@ -132,14 +132,14 @@ void BtreeAllocator::_process_range_removal(uint64_t start, uint64_t end,
 
   // | left <|////|  right |
   if (left_over && right_over) {
-    // add the spin-off right seg
-    range_seg_t seg_after{end, seg_whole.end};
-    range_tree.emplace_hint(rs, seg_after.start, seg_after.end);
-    range_size_tree.emplace(seg_after);
     // shink the left seg in offset tree
     rs->second = start;
```

Comment:
I think we should create a comment, something like:
`// btree_map.emplace_hint invalidates rs. `
otherwise there is a cryptic code move.

Reply:
I just tested that the same happens with `btree_map.emplace()` = iterator is invalidated.
Maybe we should review the code again, to check for it?

Reply:
> I think we should create a comment, something like: `// btree_map.emplace_hint invalidates rs. ` otherwise there is a cryptic code move.

done

Reply:
> I just tested that the same happens with `btree_map.emplace()` = iterator is invalidated. Maybe we should review the code again, to check for it?

I did a brief check - looks like we mostly safe, the only suspicious location is _try_remove_from_tree() function but it's not used for now since it's intended for HybridAllocator wrapping.